Minor Refactor to Prod Image Backend Render Functions

danielwrobert opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->
While digging into the backend code for the Atomic blocks, I noticed that the main render method for the Product Image block outputs a closing `</a>` tag but the opening tag is rendered by a different method.

Initially this was confusing and I thought that the closing tag was an error.

This PR is just a minor change/suggestion to render the entire anchor (opening and closing tags) via the `render_anchor` method and pass the children of the anchor to that method as args.


<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

### Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Create a post/page and add the Product Query block.
2. Confirm everything continues to work as expected:
   - No JS related to the Product Image block is loaded (the block is rendered on the server).
   - The UI of the block is identical to on the editor side.
   - The settings are reflected correctly (eg: "show on sale badge" disabled -> no on sale badge should be visible).

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

danielwrobert committed:
Refactor Product Image anchor render function.

Move closing tag for the anchor into the render function and pass badge
and image output into anchor children as args.

Also removes duplicate border radius logic from anchor render function,
as it is already present on the class render method.

danielwrobert committed:
Clean up sprintf args.

danielwrobert commented on src/BlockTypes/ProductImage.php:
```diff
@@ -115,21 +115,28 @@ private function render_on_sale_badge( $product, $attributes ) {
+	private function render_anchor( $product, $on_sale_badge, $product_image, $attributes ) {
 		$product_permalink = $product->get_permalink();
 
-		$border_radius = StyleAttributesUtils::get_border_radius_class_and_style( $attributes );
```
I also removed this `$border_radius` code from this `render_anchor` method as it does not seem to be used here.

It looks like it's handled / applied from the main `render` method below.

Unless I'm missing something?

gigitux replied:
Good point! Thanks for catching this!

gigitux reviewed (approved):
Thanks for working on this! LGTM!
I'm going pre-approving this PR, but I noticed that `woocommerce_placeholder_img_src` is a deprecated function (you can see logs on `debug.log` file).
I replaced it with `wc_placeholder_img_src` and I didn't notice any problem. Could you check this and update the PR? Thanks! 🙇

danielwrobert committed:
Replace deprecated placeholder function.

The `woocommerce_placeholder_img_src` function has been depricated since
v3.0. This replaces that function with the `wc_placeholder_img_src`
function.

danielwrobert commented:
@gigitux I tested with the `wc_placeholder_img_src` function and I'm seeing everything working as expected as well. I've updated in be2aa56. I'll go ahead and merge this.

Thanks for the review! 🚀
